cmd/node/node.go

```diff
@@ -458,14 +458,20 @@ func (app *App) initServices(ctx context.Context,
 		txs.WithLogger(app.addLogger(ConStateLogger, lg)))
 
 	var verifier blockValidityVerifier
-	msh, recovered, err := mesh.NewMesh(cdb, &verifier, app.conState, app.addLogger(MeshLogger, lg))
+	msh, err := mesh.NewMesh(cdb, &verifier, app.conState, app.addLogger(MeshLogger, lg))
 	if err != nil {
 		return fmt.Errorf("failed to create mesh: %w", err)
 	}
 
-	if !recovered {
-		if err = state.ApplyGenesis(app.Config.Genesis.ToAccounts()); err != nil {
-			return fmt.Errorf("setup genesis: %w", err)
+	genesisAccts := app.Config.Genesis.ToAccounts()
+	if len(genesisAccts) > 0 {
+		_, err = state.GetBalance(genesisAccts[0].Address)
+		if errors.Is(err, sql.ErrNotFound) {
+			if err = state.ApplyGenesis(app.Config.Genesis.ToAccounts()); err != nil {
```

Comment:
Don't know how many genesis accounts are expected, but should we not use the `genesisAccts` alrady initialized on line 466 ?